presto-hive/src/main/java/com/facebook/presto/hive/s3/PrestoS3FileSystem.java

```diff
@@ -854,8 +855,23 @@ private AWSCredentialsProvider createAwsCredentialsProvider(URI uri, Configurati
             return InstanceProfileCredentialsProvider.getInstance();
         }
 
-        if (s3IamRole != null) {
-            return new STSAssumeRoleSessionCredentialsProvider.Builder(s3IamRole, s3IamRoleSessionName).build();
+        boolean webIdentityEnabled = conf.getBoolean("presto.hive.s3.webidentity.enabled", false);
+        String webIdentityTokenFile = conf.get("presto.hive.s3.web-identity-token-file");
```

Comment:
nit: let's make these two class variables and move their initialization to the `initialize` method, where other configs are retrieved.

https://github.com/prestodb/presto/blob/f6c97505e1374e6ba064dcbfea32e5c1cbd343be/presto-hive/src/main/java/com/facebook/presto/hive/s3/PrestoS3FileSystem.java#L201-L252